lib/streamlit/runtime/secrets.py

```diff
@@ -170,26 +177,45 @@ def _parse(self, print_exceptions: bool) -> Mapping[str, Any]:
             if self._secrets is not None:
                 return self._secrets
 
-            try:
-                with open(self._file_path, encoding="utf-8") as f:
-                    secrets_file_str = f.read()
-            except FileNotFoundError:
+            # It's fine for a user to only have one secrets.toml file defined, so
+            # we ignore individual FileNotFoundErrors when attempting to read files
+            # below and only raise an exception if we weren't able read *any* secrets
+            # file.
+            found_secrets_file = False
+            secrets = {}
+
+            for path in self._file_paths:
+                try:
+                    with open(path, encoding="utf-8") as f:
+                        secrets_file_str = f.read()
+                    found_secrets_file = True
+                except FileNotFoundError:
+                    continue
+
+                try:
+                    secrets.update(toml.loads(secrets_file_str))
+                except:
+                    if print_exceptions:
+                        st.error(f"Error parsing secrets file at {path}")
+                    raise
+
+            if not found_secrets_file:
+                err_msg = f"No secrets files found. Valid paths for a secrets.toml file are: {', '.join(self._file_paths)}"
                 if print_exceptions:
-                    st.error(f"Secrets file not found. Expected at: {self._file_path}")
-                raise
+                    st.error(err_msg)
+                raise FileNotFoundError(err_msg)
 
-            try:
-                secrets = toml.loads(secrets_file_str)
-            except:
-                if print_exceptions:
-                    st.error("Error parsing Secrets file.")
-                raise
+            if len([p for p in self._file_paths if os.path.exists(p)]) > 1:
+                _LOGGER.info(
+                    f"Secrets found in multiple locations: {', '.join(self._file_paths)}. "
+                    "When multiple secret.toml files exist, local secrets will take precedence over global secrets."
+                )
```

Comment:
@sfc-gh-jcarroll heads up that I tweaked the message wording slightly because the way this code is implemented now allows us to watch an arbitrary list of `secrets.toml` files.

I think it's quite unlikely that we'll ever want to watch more than 2, but on the off chance that we do it seems worth it to me to make the error message not refer to an exact number